src/Pages/Login/Login.js

```diff
@@ -0,0 +1,169 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { Link } from "react-router-dom";
+import Card from "./Components/Card/Card";
+import EmailPasswordInput from "./Components/Input/EmailPasswordInput";
+import Buttons from "./Components/Button/Buttons";
+
+function Login() {
+  const [inputs, setInputs] = useState({
+    email: "",
+    password: "",
+  });
+
+  const [onBlurHasValue, setOnBlurHasValue] = useState(false);
+  const [buttonsEmailHasValue, setButtonsEmailHasValue] = useState(false);
+  const [buttonsPwHasValue, setButtonsPwHasValue] = useState(false);
+
+  const handleUpdateValue = (e) => {
+    const { name, value } = e.target;
+    setInputs({
+      ...inputs,
+      [name]: value,
+    });
+  };
+
+  const setOnBlurCheckValue = (e) => {
+    !email && !password ? setOnBlurHasValue(true) : setOnBlurHasValue(false);
+  };
+
+  const handleEmailCheck = (e) => {
+    e.preventDefault();
+
+    !/\S+@\S+\.\S+/.test(email)
+      ? setButtonsEmailHasValue(true)
+      : setButtonsEmailHasValue(false);
+
+    !/^.*(?=.{8,})(?=.*[0-9])(?=.*[a-zA-Z]).*$/.test(password)
+      ? setButtonsPwHasValue(true)
+      : setButtonsPwHasValue(false);
+  };
```

Comment:
```suggestion
  const handleEmailCheck = (e) => {
    e.preventDefault();
    
    const isValidEmailReg = !/\S+@\S+\.\S+/.test(email)
    const isValidPwReg = !/^.*(?=.{8,})(?=.*[0-9])(?=.*[a-zA-Z]).*$/.test(password)
    
    setButtonsEmailHasValue(isValidEmailReg)
    setButtonsPwHasValue(isValidPwReg)
  };
```
- 이런 식으로 변수 명을 명확하게 제시하면 코드를 읽을 때 보다 선언적으로 읽히는 코드가 될 것 같습니다.
- boolean 값으로 리턴된 값들을 다시 함수의 인자로 활용하면 삼항연산자를 줄일 수 있습니다.

Reply:
감사합니다 반영했습니다